apps/test/netsim/NetSimRouterNode.js

```diff
@@ -234,6 +234,110 @@ describe("NetSimRouterNode", function () {
+        router.requestAddress(wire3, 'client3', function () {});
+      });
+
+      it ("assigns passed local hostname to respective wire", function () {
+        assertEqual('client1', wire1.localHostname);
+        assertEqual('client2', wire2.localHostname);
+        assertEqual('client3', wire3.localHostname);
+      });
+
+      it ("assigns single-number addresses when using single-number address format", function () {
+        assertEqual('1', wire1.localAddress);
+        assertEqual('2', wire2.localAddress);
+        assertEqual('3', wire3.localAddress);
+      });
+
+      it ("assigns own hostname as remote hostname on all wires", function () {
+        assertEqual(router.getHostname(), wire1.remoteHostname);
+        assertEqual(router.getHostname(), wire2.remoteHostname);
+        assertEqual(router.getHostname(), wire3.remoteHostname);
+      });
+
+      it ("assigns own address (zero) as remote address on all wires", function () {
+        assertEqual('0', router.getAddress());
+        assertEqual(router.getAddress(), wire1.remoteAddress);
+        assertEqual(router.getAddress(), wire2.remoteAddress);
+        assertEqual(router.getAddress(), wire3.remoteAddress);
+      });
+    });
+
+    describe("requesting three addresses in two-part format", function () {
+      beforeEach(function () {
+        netsimGlobals.getLevelConfig().addressFormat = '4.4';
+        router.requestAddress(wire1, 'client1', function () {});
+        router.requestAddress(wire2, 'client2', function () {});
+        router.requestAddress(wire3, 'client3', function () {});
+      });
+
+      it ("assigns two-part addresses where first part is router number", function () {
+        assertEqual(router.entityID + '.1', wire1.localAddress);
+        assertEqual(router.entityID + '.2', wire2.localAddress);
+        assertEqual(router.entityID + '.3', wire3.localAddress);
+      });
+
+      it ("assigns own two-part address (router#.0) as remote address on all wires", function () {
+        assertEqual(router.entityID + '.0', router.getAddress());
+        assertEqual(router.getAddress(), wire1.remoteAddress);
+        assertEqual(router.getAddress(), wire2.remoteAddress);
+        assertEqual(router.getAddress(), wire3.remoteAddress);
+      });
+    });
+
+    describe("requesting three addresses in four-part format", function () {
+      beforeEach(function () {
+        netsimGlobals.getLevelConfig().addressFormat = '8.8.8.8';
+        router.requestAddress(wire1, 'client1', function () {});
+        router.requestAddress(wire2, 'client2', function () {});
+        router.requestAddress(wire3, 'client3', function () {});
+      });
+
+      it ("uses zeros for all except last two parts", function () {
+        assertEqual('0.0.' + router.entityID + '.1', wire1.localAddress);
+        assertEqual('0.0.' + router.entityID + '.2', wire2.localAddress);
+        assertEqual('0.0.' + router.entityID + '.3', wire3.localAddress);
```

Comment:
Three cases for each of these may seem like a bit much, since the assigned addresses are so predictable.  I've done this in part to prepare for a near-future task of randomizing address assignment, where for the tests I'll want a fixed seed but also for it to be obvious that they're assigned non-sequentially.